### PR TITLE
refactor: reduce contract size

### PR DESCRIPTION
**Pull request by nchamo**
Between #213 and #216, the contract size became too big. Now we are making it smaller, and at the same time, saving some gas 🎉

### CODE REVIEW

**Review by 0xsambugs — APPROVED** [review]
And this also made gas costs cheaper? Noice! 🎉

File: contracts/DCAHub/DCAHubPositionHandler.sol [hunk 1]
```diff
@@ -324,15 +319,23 @@ abstract contract DCAHubPositionHandler is ReentrancyGuard, DCAHubConfigHandler,
...
     }
 
-    uint256 _accumRatio = _userPosition.from < _userPosition.to
-      ? _accumRatio[_userPosition.from][_userPosition.to][_userPosition.swapIntervalMask][_newestSwapToConsider].accumRatioAToB -
-        _accumRatio[_userPosition.from][_userPosition.to][_userPosition.swapIntervalMask][_userPosition.swapWhereLastUpdated].accumRatioAToB
-      : _accumRatio[_userPosition.to][_userPosition.from][_userPosition.swapIntervalMask][_newestSwapToConsider].accumRatioBToA -
-        _accumRatio[_userPosition.to][_userPosition.from][_userPosition.swapIntervalMask][_userPosition.swapWhereLastUpdated].accumRatioBToA;
+    uint256 _positionsAccumRatio;
+    if (_userPosition.from < _userPosition.to) {
+      mapping(uint32 => AccumRatio) storage _accumRatioRef = _accumRatio[_userPosition.from][_userPosition.to][_userPosition.swapIntervalMask];
```
Comment (0xged) [inline contracts/DCAHub/DCAHubPositionHandler.sol:324]: fuck this looks awful as a declaration. btw. at the same time, wouldn't you be able to declare `_accumRatioRef` above the `if` statement, avoiding this double declaration?

File: contracts/DCAHub/DCAHubSwapHandler.sol [hunk 2]
```diff
@@ -170,13 +172,14 @@ abstract contract DCAHubSwapHandler is ReentrancyGuard, DCAHubConfigHandler, IDC
...
-      if (i > 0 && _tokens[i] <= _tokens[i - 1]) {
+      address _token = _tokens[i];
+      if (!allowedTokens[_token]) revert IDCAHubConfigHandler.UnallowedToken();
+      if (i > 0 && _token <= _tokens[i - 1]) {
         revert IDCAHub.InvalidTokens();
       }
 
       TokenInSwap memory _tokenInSwap;
-      _tokenInSwap.token = _tokens[i];
+      _tokenInSwap.token = _token;
```
Comment (0xged) [inline contracts/DCAHub/DCAHubSwapHandler.sol:182]: Nice!

**Review by 0xged — APPROVED** [review]
Amazing work!

File: contracts/DCAHub/DCAHubPositionHandler.sol (hunk 1 above)
Comment (nchamo) [reply contracts/DCAHub/DCAHubPositionHandler.sol:324]: But the thing is that they are used only inside the if, and they are different (one is from-to, and the other is to-from)

**Pull request merged**